jib-gradle-plugin-extension-api/src/main/java/com/google/cloud/tools/jib/gradle/extension/JibGradlePluginExtension.java

```diff
@@ -44,6 +60,7 @@
   ContainerBuildPlan extendContainerBuildPlan(
       ContainerBuildPlan buildPlan,
       Map<String, String> properties,
+      Optional<T> extraConfig,
```

Comment:
maybe this is just config? (vs extraConfig?) I dunno

Reply:
I am fine with `config`. I think one good effect about saying "extra" is to increase the chance of people preferring `properties` in simple cases? I can always change the name.